src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -581,10 +595,31 @@ public override uint EnableSsl(uint options)
             using (TrySNIEventScope.Create(nameof(SNIHandle)))
             {
                 _validateCert = (options & TdsEnums.SNI_SSL_VALIDATE_CERTIFICATE) != 0;
+
                 try
                 {
-                    _sslStream.AuthenticateAsClient(_targetServer, null, SupportedProtocols, false);
-                    _sslOverTdsStream.FinishHandshake();
+                    if (_isTDSS)
+                    {
+#if NETCOREAPP 
+
+                        SslClientAuthenticationOptions sslClientOptions = new()
+                        {
+                            TargetHost = _serverIndicationName,
+                            ApplicationProtocols = new List<SslApplicationProtocol>() { new(TdsEnums.TDS8) },
```

Comment:
Does this need to be a per instance collection? Is if going to be modified by the caller? if it isn't going to be modified we could have a single static instance that we reuse. If it has to be instantiated consider using the ctor that specifies the initial size because the default initial size is 5 and if we know this will almost never grow we waste some memory each time.

Reply:
can you check the new changes I made?